pkg/globalnet/controllers/ipam/gatewaymonitor_test.go

```diff
@@ -176,6 +184,52 @@ var _ = Describe("Service monitoring", func() {
 	})
 })
 
+var _ = Describe("Service Export monitoring", func() {
+	t := newTestDriver()
+
+	var service *corev1.Service
+
+	BeforeEach(func() {
+		service = newService("nginx")
+	})
+
+	JustBeforeEach(func() {
+		t.createEndpoint(newEndpointSpec(clusterID, t.hostName, localCIDR))
+	})
+
+	When("a Service without a global IP is Exported", func() {
+		It("should assign a global IP and add an appropriate IP tables rule", func() {
+			service = t.createService(service)
+			t.createSvcExport(service)
+			t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain,
+				ContainSubstring(t.awaitServiceGlobalIP(service.Name)))
+		})
+	})
+
+	When("a Service is created", func() {
+		It("should not assign a global IP or IP tables rules until service is exported", func() {
+			service = t.createService(service)
+			t.awaitNoServiceGlobalIP(service)
+			t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIPSubstr))
+
+			t.createSvcExport(service)
+			globalIP := t.awaitServiceGlobalIP(service.Name)
+			t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+		})
+	})
+
+	When("a Service Export is created before service", func() {
+		It("should assign a global IP and add an appropriate IP tables rule after service creation", func() {
+			t.createSvcExport(service)
+			t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIPSubstr))
+
+			t.createService(service)
+			globalIP := t.awaitServiceGlobalIP(service.Name)
+			t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+		})
+	})
```

Comment:
We can combine these test cases with "_Service monitoring_" since `Service` and `ServiceExport` are closely tied. Also we need a test for `ServiceExport` delete.

```
When("a Service without a global IP is created", func() {
    JustBeforeEach(func() {
        t.createEndpoint(newEndpointSpec(clusterID, t.hostName, localCIDR))
    })

    When("the Service was already exported", func() {
        It("should assign a global IP and add an appropriate IP tables rule", func() {
            t.createSvcExport(service)
            t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIPSubstr))

            t.createService(service)
            ...
        })
    })

    When("the Service is subsequently exported", func() {
        It("should eventually assign a global IP and add an appropriate IP tables rule", func() {
            service = t.createService(service)
            t.awaitNoServiceGlobalIP(service)
            t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIPSubstr))

            t.createSvcExport(service)
            ...
        })
    })
})

Context("", func() {
    JustBeforeEach(func() {
        t.createSvcExport(service)
        service = t.createService(service)
        t.createEndpoint(newEndpointSpec(clusterID, t.hostName, localCIDR))
    })

    When("a Service with a global IP is unexported", func() {
        It("should unassign its global IP and remove the appropriate IP tables rule", func() {
            globalIP := t.awaitServiceGlobalIP(service.Name, localCIDR) 
            t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))

            // delete ServiceExport
	    t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
            t.awaitNoServiceGlobalIP(service)
        })
    })

    When("a Service's global IP is updated", func() { ... the rest of the tests
})
```

Reply:
Yeah, was planning to merge them once I figured out test cases and got them working. Find it easier to write separate and just run that test case when writing new ones. Thanks for easy to copy-paste refactor :D